app/core/src/main/java/com/fsck/k9/mailstore/LocalMessage.java

```diff
@@ -41,6 +56,12 @@
     private PreviewType previewType;
     private boolean headerNeedsUpdating = false;
 
+    // Only available for messages in Outbox folder after calling fetchOutboxState()
```

Comment:
this looks pretty ugly to me. could this (and related methods) be kept in an OutboxStateRepository or something? the calls are simple and rare enough that we don't need to worry about caching, plus it would be more easily testable.